### PR TITLE
Add a wrapper for a discord message and discord source

### PR DESCRIPTION
**Pull request by CurvaXa**
Goal: 
To make commands reusable from the different sources (e.g. slack) we need to decouple command and command source and source-specific messages.

Partially tested (!ping & !help works ;)

**Semantic versioning classification:**
- [x] This PR changes the core classes (e.g. base classes like Command, BotTable etc.)

### CODE REVIEW

File: src/commands_discord/social/remind-command.js [hunk 1]
```diff
@@ -105,7 +105,7 @@ class RemindCommand extends DiscordCommand {
   /**
    * Gets the default value for a given argument definition.
    * Used when unable to scan the argument from the command's text.
-   * @param  {Message}        message the command's message
+   * @param  {BaseMessage}        message the command's message
    * @param  {CommandArgDef}  arg     the argument definition
    * @return {Object}                 the default value
    */
```
Comment (alteh-union) [inline src/commands_discord/social/remind-command.js:111]: This function is broken, because you call "channel.id" on the BaseMessage instead of the originalMessage. Please check "getDefaultDiscordArgValue" function in all commands and change "message" to "message.originalMessage" where needed.

File: src/components/base-message.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    this.originalMessage = originalMessage;
+
+  }
+
+  static createFromSlack(message) {
+    return new BaseMessage(message.team_id, message.channel, message.user, message.text, message);
+  }
+
+  static createFromDiscord(message) {
+    return new BaseMessage(message.guild.id, message.channel.id, message.author.id, message.content, message);
```
Comment (alteh-union) [inline src/components/base-message.js]: This throws an error on trying to send a DM to the Bot, because DM messages do not have any guild associated.

Suggesting to check if message.guild is defined, and if not, then just pass "null" as the value.

Also suggest to pass "null" in "createFromSlack" if "message.team_id" is undefined.

File: src/managers/commands-parser.js [hunk 3]
```diff
@@ -137,32 +137,24 @@ class CommandsParser {
...
-    const currentUserLocale = await this.context.dbManager.getUserSetting(
-      message.source.name,
-      message.teamId,
-      message.userId,
-      UserSettingsTable.USER_SETTINGS.localeName.name
-    );
+    const commandLangManager = await this.getCommandLangManager(message);
+    const command = this.parseMessage(message, currentPrefix, commandLangManager);
+    if (command) {
+      return await this.executeCommand(message, command, commandLangManager);
```
Comment (alteh-union) [inline src/managers/commands-parser.js:70]: This function must return whether a command was found or not. Now you return nothing, basically, so when this function finishes, the Bot has no chance to understand whether it's a command or just a regular text message and falls into line 108 in index.js file ("c.messageModerator.premoderateDiscordMessage(message);").

Because of that - it will try to premoderate the message even though it was a command to the Bot. For example, if you try to add some bad words while censoring is enabled, it immediately will remove your command message.

Basically, I think that Bot commands should avoid premoderation, so it's better to return something like (command !== null) from here.

**Comment by CurvaXa** [comment]
@alteh-union verify & merge please, Actually, u can make a squash & merge, because all this chages are just refactoring.
